### ISSUE TITLE
False positive on `EqualsWithHashCodeExist`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6569]
## Expected Behavior
Below TC should pass when run in file `EqualsWithHashCodeExistSpec`

```kotlin
@Test
fun `1`() {
    val code = """
        open class Base1 {
            open fun Base1.equals(other: Any?): Boolean {
                return false
            }
        }
        class A1 : Base1() {
            override fun Base1.equals(other: Any?): Boolean {
                return true
            }
        }
    """.trimIndent()
    assertThat(subject.compileAndLint(code)).isEmpty()
}
```

## Observed Behavior
Above TC is failing

## Context
This is kind of reversed reversed-engineered example but I think we should fix this. This will be fixed in a common method(`isEqualsFunction`) defined in the `MethodSignature.kt` file which is used at more places fixing it there will ensure no wrong behaviour at other rules as well
